packages/core/src/apiml/Services.ts

```diff
@@ -0,0 +1,244 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, IConfig, IConfigProfile, ImperativeConfig, ImperativeExpect, Logger,
+         PluginManagementFacility, RestClient } from "@zowe/imperative";
+import { ApimlConstants } from "./ApimlConstants";
+import { IApimlProfileInfo } from "./doc/IApimlProfileInfo";
+import { IApimlService } from "./doc/IApimlService";
+import { IApimlSvcAttrsLoaded } from "./doc/IApimlSvcAttrsLoaded";
+import * as JSONC from "comment-json";
+import { ConfigConstants } from "../../../../../imperative/lib";
+
+/**
+ * Class to handle listing services on APIML gateway.
+ * @export
+ * @class Services
+ */
+export class Services {
+    /**
+     * Forms a list of APIML service attributes needed to query APIML for every
+     * REST service for every loaded command group. This information can later
+     * be used to create every connection profile required for every loaded
+     * command group.
+     *
+     * @returns The APIML service attributes needed to query APIML.
```

Comment:
Should we document the errors that this function can throw (e.g. `@throw [XYZ error] when loadedConfig is not defined. Call Imperative.init()`)?
The `[XYZ error]` portion may not be necessary since they will all be `ImperativeError`s